include/tvm/relay/dataflow_functor.h

```diff
@@ -0,0 +1,248 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
```

Comment:
naming: perhaps it should be `dataflow_pattern_functor.h`? since `dataflow_functor` is a bit confusing